serverless-workflow-operator/pkg/controller/workflow/workflow_controller.go

```diff
@@ -320,3 +342,56 @@ func getImage(cr *v1.Workflow) string {
 func intPtr(x int32) *int32 {
 	return &x
 }
+
+func newKnativeService(cr *v1.Workflow) *knsv1.Service {
+	return &knsv1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      cr.Name,
+			Namespace: cr.Namespace,
+			Labels:    getLabels(cr),
+		},
+		Spec: knsv1.ServiceSpec{
+			ConfigurationSpec: knsv1.ConfigurationSpec{
+				Template: knsv1.RevisionTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      cr.Name + "-v1",
+						Namespace: cr.Namespace,
+						Labels:    getDeploymentLabels(cr),
+					},
+					Spec: knsv1.RevisionSpec{
+						PodSpec: corev1.PodSpec{
```

Comment:
You can create a common PodSpec and add it there and also to the v1.Service

Reply:
will do

Reply:
Knative doesn't support env with downward api (https://github.com/knative/serving/issues/4190).
I'll use value `cr.Namespace` for the `NAMESPACE` env.